packages/node/src/integrations/tracing/vercelai/instrumentation.ts

```diff
@@ -236,13 +245,18 @@ export class SentryVercelAiInstrumentation extends InstrumentationBase {
         };
 
         return handleCallbackErrors(
-          async () => {
+          () => {
             // @ts-expect-error we know that the method exists
-            const result = await originalMethod.apply(this, args);
+            const result = originalMethod.apply(this, args);
 
-            // Tool errors are not rejected in Vercel V5, it is added as metadata to the result content
-            checkResultForToolErrors(result);
+            if (isPromiseLike(result)) {
+              // check for tool errors when the promise resolves, keep the original promise identity
+              result.then(checkResultForToolErrors, () => {});
```

Comment:
hmm, I think we need to re-throw here or something like this, as otherwise this could swallow unhandled promise rejections, possibly...? Can this be? This is always hard tor reason about for me...